config/store.go

```diff
@@ -175,9 +179,13 @@ func (s *Store) Set(newCfg *model.Config) (*model.Config, error) {
 		return nil, ErrReadOnlyStore
 	}
 
-	oldCfg := s.config.Clone()
+	newCfg = newCfg.Clone()
```

Comment:
It is not entirely clear to me the need to clone here. Is it out of safety or there is a reason for this?

Reply:
Yes, as far as I could see I've added it only out of safety and respect to the caller's data.